src/DebuggerForm.cpp

```diff
@@ -1537,6 +1591,50 @@ void DebuggerForm::processMerge(const QString& message)
 	}
 }
 
+QByteArray DebuggerForm::saveCommands() const
+{
+	QByteArray output;
+
+	auto first = true;
+	for (const auto& [name, description, source, icon, _] : commands) {
+		if (!first) output.append(':');
+		first = false;
+		output.append(name.toUtf8().toBase64());
+		output.append(';');
+		output.append(description.toUtf8().toBase64());
+		output.append(';');
+		output.append(source.toUtf8().toBase64());
+		output.append(';');
+		output.append(icon.toUtf8().toBase64());
+	}
+
+	return output;
+}
+
+void DebuggerForm::restoreCommands(const QByteArray &input)
+{
+	commands.clear();
+
+	auto commandBytes = input.split(':');
+	auto counter = 0;
+	for (const auto& cb: commandBytes) {
+		if (cb.isEmpty()) continue;
```

Comment:
This is fine. You could also leave it out, and rely on the `fields.count() < 4` check below.  (I don't know how common empty lines are).